### PR TITLE
Post process md links in package-list files.

### PR DESCRIPTION
**Pull request by yschimke**
Fix the Dokka -> Mkdocs pipeline with md.

#fixes https://github.com/square/okhttp/issues/7338#issuecomment-1221245181

### CODE REVIEW

**Review by JayNewstrom — APPROVED** [review]
Thanks!

File: deploy_website.sh [hunk 1]
```diff
@@ -34,6 +34,10 @@ cd $DIR
   :okhttp-urlconnection:dokkaGfm \
   :okhttp:dokkaGfm
 
+# mkdocs doesn't correctly handle the .md paths in package-list, so post-process.
+# See https://github.com/square/okhttp/issues/7338
+find docs/4.x -name package-list -exec sed -i '' -e s/index.md$// -e s/\.md$/\\//g {} \;
```
Comment (colinrtwhite) [inline deploy_website.sh:39]: Looks good, but I would tweak it slightly to do:

```
find docs/4.x -name package-list -exec sed -i '' -e 's/index.md$/index.html/' -e 's/\.md$/\.html/' -e '1 s/gfm/html/' -e '2 s/md/html/' {} \;
```

That keeps the `index.html` and also updates the first two lines to remove references to `gfm` and `md`. After those changes it produces the same text as my [manually updated `package-list`](https://colinwhite.me/okhttp3-package-list).

**Review by colinrtwhite — APPROVED** [review]

File: deploy_website.sh (hunk 1 above)
Comment (yschimke) [inline deploy_website.sh:39]:
```suggestion
find docs/4.x -name package-list -exec sed -i '' -e 's/index.md$/index.html/' -e 's/\.md$/\.html/' -e '1 s/gfm/html/' -e '2 s/md/html/' {} \;
```

**Review by swankjesse — COMMENTED** [review]
LGTM

**Pull request merged**